wazo_call_logd/tests/test_generator_scenarios.py

```diff
@@ -0,0 +1,424 @@
+    tenant_uuid: str | None = None,
+) -> dict:
+    """
+    Mocks a line.
+
+    :param id: The line ID.
+    :param name: The line name.
+    :param protocol: The line protocol.
+    :param users: The users associated with the line.
+    :param context: The line context.
+    :param extensions: The line extensions.
+    :param tenant_uuid: The tenant UUID.
+    :return: The mocked line.
+    """
+    return mock_dict(
+        {
+            'id': id,
+            'name': name,
+            'protocol': protocol,
+            'users': users or [],
+            'context': context,
+            'extensions': extensions or [],
+            'tenant_uuid': tenant_uuid,
+        }
+    )
+
+
+def mock_switchboard(uuid: str, name: str | None = None) -> dict:
+    """
+    Mocks a switchboard.
+
+    :param uuid: The switchboard UUID.
+    :param name: The switchboard name.
+    :return: The mocked switchboard.
+    """
+    return mock_dict({'uuid': uuid, 'name': name})
+
+
+def mock_context(id: int, name: str, tenant_uuid: str) -> dict[str, int | str]:
+    """
+    Mocks a context.
+
+    :param id: The context ID.
+    :param name: The context name.
+    :param tenant_uuid: The tenant UUID.
+    :return: The mocked context.
+    """
+    return mock_dict({'id': id, 'name': name, 'tenant_uuid': tenant_uuid})
+
+
+def mock_confd_client(
+    lines: list[dict] = None,
+    users: list[dict] = None,
+    contexts: list[dict] = None,
+) -> ConfdClient:
+    confd_client = create_autospec(
+        ConfdClient, instance=True, lines=Mock(), users=Mock(), contexts=Mock()
+    )
+    confd_client.lines.list.side_effect = lambda name=None, **kwargs: {
+        'items': [line for line in lines if line['name'] == name]
+        if lines and name
+        else lines
+        if lines and not name
+        else []
+    }
+    confd_client.users.get.side_effect = (
+        (lambda uuid: next(user for user in users if user['uuid'] == uuid))
+        if users is not None
+        else HTTPError
+    )
+    confd_client.contexts.list.side_effect = lambda name=None, **kwargs: {
+        'items': [context for context in contexts if context['name'] == name]
+        if name and contexts
+        else contexts
+        if contexts
+        else []
+    }
+    return confd_client
+
+
+class TestCallLogGenerationScenarios(TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        logging.basicConfig(
+            level=logging.DEBUG,
+            format='%(levelname)s: %(name)s:%(lineno)d : %(funcName)s: %(message)s',
+        )
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.confd_client = mock_confd_client()
+        self.generator = CallLogsGenerator(
+            self.confd_client,
+            [
+                LocalOriginateCELInterpretor(),
+                DispatchCELInterpretor(
+                    CallerCELInterpretor(),
+                    CalleeCELInterpretor(),
+                ),
+            ],
+        )
```

Comment:
This is duplicated hence brittle. If we add an additional interpretor, the tests could miss some errors. We should probably extract it to a function so that tests and production code run with the same parameters.